Source/Plugins/CoherenceViewer/CoherenceNode.cpp

```diff
@@ -174,13 +174,35 @@ void CoherenceNode::run()
     }
 }
 
-void CoherenceNode::updateDataBufferSize()
+void CoherenceNode::updateDataBufferSize(int newSize)
 {
-    dataWriter->clear();
-    for (int i = 0; i < nGroup1Chans + nGroup2Chans; i++)
+    int totalChans = nGroup1Chans + nGroup2Chans;
+
+    // no writers or readers can exist here
+    // so this probably can't be called during acquisition
+    // I changed it because it doesn't really make sense to just change the buffer we're currently
+    // writing to when acquisition is stopped, and we need to figure out how to
+    // make changes when acquisition is running anyway.
+    dataBuffer.apply([=](Array<FFTWArray>& arr)
     {
-        dataWriter->add(std::move(FFTWArray(segLen * Fs)));
-    }
+        for (int i = 0; i < jmin(totalChans, arr.size()); i++)
+        {
+            arr.getReference(i).resize(newSize);
+        }
+
+        int nChansChange = totalChans - arr.size();
+        if (nChansChange > 0)
+        {
+            for (int i = 0; i < nChansChange; i++)
+            {
+                arr.add(FFTWArray(newSize));
```

Comment:
You don't actually need `move` here since you're passing in a temporary, which is already an rvalue. (not sure how far down the move semantics rabbithole you've gone yet..)

Reply:
Alright I'll have to wrap my head around this change. Definitely have not gone as far down as you!